vault/mount.go

```diff
@@ -685,7 +685,14 @@ func (c *Core) mountInternal(ctx context.Context, entry *MountEntry, updateStora
 			locked = false
 		}
 	}
-	defer unlock()
+
+	return c.mountInternalWithLock(ctx, entry, updateStorage, unlock)
+}
+
+func (c *Core) mountInternalWithLock(ctx context.Context, entry *MountEntry, updateStorage bool, unlock func()) error {
```

Comment:
@phil9909 @wslabosz-reply Not sure that I like this as a potential fix, plus it causes other failures.

Thoughts?

Reply:
I think we'll pull this and let #2040 merge instead, for GA.